File: python/svg_processor.py
#!/usr/bin/env python3
"""
SVG+Python Processor - Wykonuje pliki SVG z wbudowanym kodem Python
Podobnie jak router.php, ale dla Python

Użycie:
    python svg_processor.py <plik.svg>
    python svg_processor.py todo-manager-python.svg
"""

import sys
import os
import re
from datetime import datetime
import subprocess

def process_svg_file(svg_file):
    """Przetwarza plik SVG z wbudowanym kodem Python"""
    
    if not os.path.exists(svg_file):
        print(f"Błąd: Plik '{svg_file}' nie istnieje")
        return False
    
    if not svg_file.endswith('.svg'):
        print(f"Błąd: Plik musi mieć rozszerzenie .svg")
        return False
    
    # Przygotuj zmienne Python dostępne w SVG
    current_time = datetime.now().strftime("%H:%M:%S")
    current_date = datetime.now().strftime("%Y-%m-%d")
    tasks_count = 7  # Przykładowa wartość
    
    # print(f"<!-- SVG+Python renderowany przez CLI: {svg_file} o {current_time} -->")
    
    try:
        # Wczytaj plik SVG
        with open(svg_file, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Znajdź i wykonaj kod Python między <?python ... ?>
        def replace_python_code(match):
            python_code = match.group(1)
            
            # Utwórz namespace z dostępnymi zmiennymi
            namespace = {
                'current_time': current_time,
                'current_date': current_date,
                'tasks_count': tasks_count,
                'datetime': datetime,
                'os': os,
                'sys': sys
            }
            
            # Wykonaj kod Python i zachwyć output
            old_stdout = sys.stdout
            from io import StringIO
            sys.stdout = StringIO()
            
            try:
                exec(python_code, namespace)
                result = sys.stdout.getvalue()
            except Exception as e:
                result = f"<!-- Python Error: {e} -->"
            finally:
                sys.stdout = old_stdout
            
            return result
        
        # Zamień kod Python na wyniki
        processed_content = re.sub(r'<\?python\s*(.*?)\s*\?>', replace_python_code, content, flags=re.DOTALL)
        
        # Wyświetl przetworzoną zawartość
        print(processed_content)
        
        return True
        
    except Exception as e:
        print(f"Błąd podczas przetwarzania pliku: {e}")
        return False

def main():
    if len(sys.argv) != 2:
        print("Użycie: python svg_processor.py <plik.svg>")
        print("Przykład: python svg_processor.py todo-manager-python.svg")
        sys.exit(1)
    
    svg_file = sys.argv[1]
    
    if not process_svg_file(svg_file):
        sys.exit(1)

if __name__ == "__main__":
    main()

File: nodejs/svg_processor.js
#!/usr/bin/env node
/**
 * SVG+Node.js Processor - Wykonuje pliki SVG z wbudowanym kodem JavaScript
 * Podobnie jak router.php, ale dla Node.js
 * 
 * Użycie:
 *   node svg_processor.js <plik.svg>
 *   node svg_processor.js todo-manager-nodejs.svg
 */

const fs = require('fs');
const path = require('path');

function processSvgFile(svgFile) {
    if (!fs.existsSync(svgFile)) {
        console.error(`Błąd: Plik '${svgFile}' nie istnieje`);
        return false;
    }

    if (!svgFile.endsWith('.svg')) {
        console.error(`Błąd: Plik musi mieć rozszerzenie .svg`);
        return false;
    }

    // Przygotuj zmienne JavaScript dostępne w SVG
    const currentTime = new Date().toLocaleTimeString('pl-PL', { 
        hour: '2-digit', 
        minute: '2-digit', 
        second: '2-digit' 
    });
    const currentDate = new Date().toLocaleDateString('pl-PL');
    const tasksCount = 8; // Przykładowa wartość
    const nodeVersion = process.version;
    
//    console.log(`<!-- SVG+Node.js renderowany przez CLI: ${svgFile} o ${currentTime} -->`);
    
    try {
        // Wczytaj plik SVG
        const content = fs.readFileSync(svgFile, 'utf8');
        
        // Znajdź i wykonaj kod JavaScript między <?js ... ?>
        const processedContent = content.replace(
            /<\?js\s*([\s\S]*?)\s*\?>/g,
            (match, jsCode) => {
                try {
                    // Utwórz kontekst z dostępnymi zmiennymi
                    const sandbox = {
                        currentTime,
                        currentDate,
                        tasksCount,
                        nodeVersion,
                        Date,
                        Math,
                        console,
                        JSON,
                        // Capture output
                        output: '',
                        print: function(text) {
                            this.output += text;
                        }
                    };
                    
                    // Prostsza metoda - użyj eval w kontrolowanym kontekście
                    let output = '';
                    const print = (text) => { output += text; };
                    
                    // Wykonaj kod JavaScript z dostępem do zmiennych
                    eval(jsCode);
                    
                    const result = output;
                    
                    return result || '';
                    
                } catch (error) {
                    return `<!-- JavaScript Error: ${error.message} -->`;
                }
            }
        );
        
        // Wyświetl przetworzoną zawartość
        console.log(processedContent);
        
        return true;
        
    } catch (error) {
        console.error(`Błąd podczas przetwarzania pliku: ${error.message}`);
        return false;
    }
}

function main() {
    const args = process.argv.slice(2);
    
    if (args.length !== 1) {
        console.log('Użycie: node svg_processor.js <plik.svg>');
        console.log('Przykład: node svg_processor.js todo-manager-nodejs.svg');
        process.exit(1);
    }
    
    const svgFile = args[0];
    
    if (!processSvgFile(svgFile)) {
        process.exit(1);
    }
}

if (require.main === module) {
    main();
}
